queue/bytes_queue.go

```diff
@@ -41,6 +39,21 @@ type queueError struct {
 	message string
 }
 
+// getUvarintSize returns the number of bytes to encode x in uvarint format
+func getUvarintSize(x uint32) int {
```

Comment:
Where the numbers are taken from https://github.com/mohae/uvarint/blob/master/uvarint.go#L61-L69

Reply:
You should check out the code of go standard library encoding/binary, I have not used this library in my codes.

Reply:
https://play.golang.org/p/wYubPt7pY79

Reply:
Please note the difference between ``PutUvarint`` and ``PutVarint``

Reply:
https://play.golang.org/p/s9fuPeIt7o4

Reply:
That's interesting https://en.wikipedia.org/wiki/Find_first_set

Reply:
> Please note the difference between `PutUvarint` and `PutVarint` 

@neal-zhu :+1:  My bad :)